lib/posix-process/events.c

```diff
@@ -0,0 +1,161 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Copyright (c) 2022, NEC Laboratories Europe GmbH, NEC Corporation.
+ *                     All rights reserved.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/init.h>
+#include <uk/process.h>
+
+UK_EVENT(POSIX_PROCESS_CLONE_EVENT);
+UK_EVENT(POSIX_PROCESS_EXECVE_EVENT);
+UK_EVENT(POSIX_PROCESS_PPEXIT_EVENT); /* pprocess */
+UK_EVENT(POSIX_PROCESS_PTEXIT_EVENT); /* pthread */
+UK_EVENT(POSIX_PROCESS_RELEASE_EVENT);
+
+extern const struct posix_process_clonetab_entry _posix_process_clonetab_start[];
+extern const struct posix_process_clonetab_entry _posix_process_clonetab_end;
+
+#define posix_process_clonetab_foreach(itr)				\
+	for ((itr) = DECONST(struct posix_process_clonetab_entry*,	\
```

Comment:
Very minor nit. I just realised, but do we actually need this to not be const? (the iterator could be const as well)

Reply:
Honestly I took it as is from the existing implementation without thinking much about it.